CMakeLists.txt

```diff
@@ -1,6 +1,20 @@
 cmake_minimum_required(VERSION 2.8)
 project(BADVPN C)
 
+if(ANDROID)
+    set(BUILD_NOTHING_BY_DEFAULT 1 CACHE INTERNAL "BUILD_NOTHING_BY_DEFAULT" )
+    set(BUILD_TUN2SOCKS 1 CACHE INTERNAL "BUILD_TUN2SOCKS" )
+    set(HAVE_SYSLOG_H 1 CACHE INTERNAL "HAVE_SYSLOG_H" )
+
+    set(CMAKE_C_FLAGS  "${CMAKE_C_FLAGS} -Wall -Wextra -std=gnu99")
+    include_directories(${LIB_ANCILLARY_DIR})
+
+    add_definitions(-DBADVPN_THREADWORK_USE_PTHREAD -DBADVPN_LINUX -DBADVPN_BREACTOR_BADVPN -D_GNU_SOURCE)
+    add_definitions(-DBADVPN_USE_SIGNALFD -DBADVPN_USE_EPOLL)
+    add_definitions(-DBADVPN_LITTLE_ENDIAN -DBADVPN_THREAD_SAFE)
+    add_definitions(-DNDEBUG -DANDROID -D__ANDROID__)
```

Comment:
By the way, `ANDROID` and `__ANDROID__` should be defined by NDK toolchain.

Reply:
removed.